### DISCUSSION TITLE
Request: New Sources

### DISCUSSION THREAD

**Discussion by rumboalla** [rumboalla/apkupdater#394] [General]
Ordered by importance.

* Play store (with bundle support)
* ~~Gitlab~~, Hosted GitLabs.
* Online retrieval of the list of GitHub/GitLab apps.
* Codeberg
* Custom F-Droid repos. Support F-Droid mirrors.
* ~~ApkPure~~
* Uptodown
* RSS sources
* Source with support for custom list of urls.
* Apkcombo
* Apkflash
* Galaxy Store

**Comment by VasilisPngs** [comment]
Add and Apkcombo.

**Comment by esdreesh** [comment]
[Request] Add "apkflash.com" as update source. Thanks

**Comment by rumboalla** [comment]
> Add and Apkcombo.
> [Request] Add "apkflash.com" as update source. Thanks

Added both sources to the list, but bear in mind that sources that require scrapping are very unlikely to ever be added.

**Comment by benliberti** [comment]
Please could you add APKPure as a source. I've still got your old app installed as it supports APKPure.
Love the new UI.

**Comment by KingCuffey** [comment]
For some reason it's not showing the correct updates.. not even ones that are on the playstore..?

**Comment by peat80** [comment]
Uptodown should be added to the list.

**Reply by TPS** [reply]
Please note https://github.com/rumboalla/apkupdater/discussions/396#discussioncomment-6730153 & https://github.com/rumboalla/apkupdater/discussions/394#discussioncomment-6738298

**Comment by crse** [comment]
Codeberg repo support would be a great addition :)

**Reply by rumboalla** [reply]
Added to the list

**Comment by Pantyhose-X** [comment]
Play store: Get a screenshot of the app update from the Play store web, and then you should download it using the [apkpure downloader](https://apkpure.com/apk-downloader). 
The downloader code should be integrated into Apkupdater, not via [apkpure downloader](https://apkpure.com/apk-downloader).

**Comment by samuraikid0** [comment]
I think we should default
 git, fdroid, Izzy and apkcombo 
once implemented since those aren't behind cuckflare

**Comment by TotallyAvailable** [comment]
Also maybe a filter to reduce available per App updates to one "most recent/preferred" (Ideally linked to a Preferred Source Option to avoid Signature conflicts) entry as even with only GitHub and APKMirror enabled and increasing GitHub coverage, double entries are a frequent occurrence, getting worse with expanding Source selection of course.

How much of a pain that's going to be... following related Discussion over at Obtainium (Edit for more context: https://github.com/ImranR98/Obtainium/issues/946) on improving Version Detections, at least an only show Update x from Source y Option (= If a GitHub entry exists -> don't show any other selected source as well). There's that one App I'm thinking about that kinda (not actually tried it) does that (UpgradeAll)...Also more of a feature request than a New Source suggestion, still a related issue.

**Comment by vipulm7** [comment]
Google Play Store support!

**Reply by TPS** [reply]
That's covered extremely well by [Aurora Store](https://f-droid.org/packages/com.aurora.store) & fraught w/ legalities in various places.  _I'd_ be ok w/ keeping this separated out.

**Reply by TPS** [reply]
Also, it's already the 1st item on the OP list. 🤷🏾‍♂️

**Comment by Jvr2022** [comment]
I am currently working at my local pc for making codeberg support

**Reply by Jvr2022** [reply]
api: codeberg.org/api/v1

**Reply by Jvr2022** [reply]
#525

**Comment by electriquo** [comment]
i've tried to analyze `https://galaxystore.samsung.com/detail/<package name>`. it seems that galaxy store can be easily supported....
given a package , e.g. `com.samsung.android.oneconnect`

version can be retrieved by:

```shell
$ curl -Ls 'https://galaxystore.samsung.com/api/detail/com.samsung.android.oneconnect' | jq -r '.DetailMain.contentBinaryVersion'
1.8.13.23
```

pretty name can be retrieved by:

```shell
$ curl -Ls 'https://galaxystore.samsung.com/api/detail/com.samsung.android.oneconnect' | jq -r '.DetailMain.contentName'
SmartThings
```

deep link can be retrieved by:

```shell
$ curl -Ls 'https://galaxystore.samsung.com/api/detail/com.samsung.android.oneconnect' | jq -r '.DetailMain.deeplinkUrl'
samsungapps://ProductDetail/com.samsung.android.oneconnect?session_id=W_4292513AC7053E29D0655B06CBB372D9
```

i assume that if there is an update, then galaxy store app can be opened using the deeplink and the user can click "update".


what do you think @rumboalla?

**Reply by jzallas** [reply]
I took these suggestions and just wrote a quick shell script to update my own device
https://gist.github.com/jzallas/c176c161c9fc3fc9702eb8ea71f0ac47#file-check_update-sh

It didn't catch _everything_ though, as the samsung store still says i have 5 apps that need updating and I refuse to sign into a samsung account to find out which 5 it is.

I will share that you probably need a couple more query parameters with that first curl describing the device you're trying to update. As-is, it will often tell you there is an update but the deeplink on your device will say it's incompatible

**Comment by ghost** [comment]
add
- https://www.rustore.ru/catalog
> Rustore has many apps that other app stores don't have.

- http://m.appchina.com/soft
> China Developer's App Store.
